subsys/bluetooth/controller/hci/nordic/hci_vendor.c

```diff
@@ -9,7 +9,7 @@
 #include <bluetooth/addr.h>
 #include <bluetooth/hci_vs.h>
 
-#include <nrf.h>
+#include <nrfx.h>
```

Comment:
It would be nice to add a comment here, why `<nrfx.h>` is in fact needed. People may get confused, as the code that follows does not justifies it, and for building for a real SoC `<nrf.h>` suffices.

Reply:
@anangl . What about: `// Include nrfx.h to make it portable to simulated SoCs as well`.
We may even consider to not add the mdk folder in the include paths

Reply:
On second thought, the problem seems to be even worse. It's not only about confusion. In the current form, when building for nrf52_bsim, depending on whether `<nrf.h>` or `<nrfx.h>` is included from a given source file, you'll get different base addresses of peripherals there. This may lead to hard to find issues when, for example, a peripheral base address passed from one module will be compared with a base address of a specific instance of that peripheral (let's say to check if some capabilities are available) in some other module.
I think a solution for this could be to provide an `nrf.h` file in `HW_models/` that would only include `<mdk/nrf.h>` and redefine the base address (instead of doing it in `nrfx_bsim_redef.h`). Then enforce this `nrf.h` file to be always included first, by adding `HW_models/` at the beginning of include paths.

> We may even consider to not add the mdk folder in the include paths

Yes, we could include MDK files with `<mdk/...>`, but the problem I described above would sill exist. One could still include `<mdk/nrf.h>` in some source file where only the base address of some peripheral is needed.

Reply:
I've added the mocked nrf.h and removed the commit changing "include nrf.h -> nrfx.h". Now we can again include nrf.h, given that the nrf.h provided by bsim is included first. That, is, as it was before.